src/NHibernate/Dialect/MsSql2000Dialect.cs

```diff
@@ -352,6 +352,7 @@ protected virtual void RegisterFunctions()
 			// Casting to CHAR (without specified length) truncates to 30 characters. 
 			// A longer version would be safer, but 50 is enough to prevent errors when casting uniqueidentifer to a string representation (NH-2858)
 			RegisterFunction("str", new SQLFunctionTemplate(NHibernateUtil.String, "cast(?1 as nvarchar(50))"));
```

Comment:
Off-topic side-note: I have missed that hard-coded `50` length limitation in a cast. It should use the default cast length instead. Worth a PR?
(Of course for the `strguid` case, using a hard-coded length make more sense.)

Reply:
I think so, after this issue I can create a new PR remove this hard-code.